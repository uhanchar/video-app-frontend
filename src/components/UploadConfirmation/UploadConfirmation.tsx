import React, { FC } from 'react';
import { Snackbar, IconButton } from '@material-ui/core';
import { Close as CloseIcon } from '@material-ui/icons';

export interface IUploadConfirmation {
  isOpen: boolean;
  closeSnackbar: () => void;
}

const UploadConfirmation: FC<IUploadConfirmation> = ({ isOpen, closeSnackbar }) => {
  const handleClose = () => {
    closeSnackbar();
  };

  return (
    <Snackbar
      anchorOrigin={{
        vertical: 'bottom',
        horizontal: 'left',
      }}
      open={isOpen}
      autoHideDuration={6000}
      data-testid="close-confirmation"
      onClose={handleClose}
      message="Video was successfully uploaded"
      action={(
        <IconButton
          size="small"
          aria-label="close"
          color="inherit"
          data-testid="close-confirmation-button"
          onClick={handleClose}
        >
          <CloseIcon fontSize="small" />
        </IconButton>
      )}
    />
  );
};

export default UploadConfirmation;
