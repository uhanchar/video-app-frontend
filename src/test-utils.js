import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { applyMiddleware, createStore } from 'redux';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import thunk from 'redux-thunk';

import rootReducer from './reducers';

const reducerInitialState = {};

function render(
  ui,
  {
    initialState = reducerInitialState,
    store = createStore(rootReducer, initialState, applyMiddleware(thunk)),
    ...renderOptions
  } = {},
) {
  // eslint-disable-next-line react/prop-types
  function Wrapper({ children }) {
    return (
      // eslint-disable-next-line react/jsx-filename-extension
      <Provider store={store}>
        <BrowserRouter>
          {children}
        </BrowserRouter>
      </Provider>
    );
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from '@testing-library/react';

export { render };
