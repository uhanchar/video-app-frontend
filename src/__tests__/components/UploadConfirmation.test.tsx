import React from 'react';
import { fireEvent } from '@testing-library/react';

import { render as renderWithRedux } from 'test-utils';
import UploadConfirmation from 'components/UploadConfirmation/UploadConfirmation';

const closeSnackbar = jest.fn();

describe('test for UploadConfirmation component', () => {
  it('should render Loader component', () => {
    const { asFragment } = renderWithRedux(<UploadConfirmation isOpen closeSnackbar={closeSnackbar} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should call closeSnackbar on close button click', () => {
    const { getByTestId } = renderWithRedux(<UploadConfirmation isOpen closeSnackbar={closeSnackbar} />);

    fireEvent.click(getByTestId('close-confirmation-button'));

    expect(closeSnackbar).toHaveBeenCalled();
  });
});
