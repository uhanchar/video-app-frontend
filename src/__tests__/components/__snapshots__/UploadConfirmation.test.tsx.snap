// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for UploadConfirmation component should render Loader component 1`] = `
<DocumentFragment>
  <div
    class="MuiSnackbar-root MuiSnackbar-anchorOriginBottomLeft"
    data-testid="close-confirmation"
  >
    <div
      class="MuiPaper-root MuiSnackbarContent-root MuiPaper-elevation6"
      direction="up"
      role="alert"
      style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <div
        class="MuiSnackbarContent-message"
      >
        Video was successfully uploaded
      </div>
      <div
        class="MuiSnackbarContent-action"
      >
        <button
          aria-label="close"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall"
          data-testid="close-confirmation-button"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
